src/main/webapp/app/entities/competency/learning-path.model.ts

```diff
@@ -34,6 +35,30 @@ export class NgxLearningPathNode implements Node {
     public label?: string;
 }
 
+export function getIcon(node: NgxLearningPathNode) {
+    // return generic icon if no type present
+    if (!node.type) {
+        return faCircle;
+    }
+
+    // return different icons for completed learning objects
+    if (node.type === NodeType.EXERCISE || node.type === NodeType.LECTURE_UNIT) {
+        if (node.completed) {
+            return faCheckCircle;
+        } else {
+            return faPlayCircle;
+        }
+    }
+
+    const icons = {
```

Comment:
Why not use a switch here?

Reply:
This follows the pattern from other getIcon Methods (lecture units, exercises, competencies). Also, with formatting a switch takes up double the amount of lines and makes it more difficult to read imo.

Reply:
Why doesn't JavaScript have a switch expression that you can immediately return 😢
Then it's fine to me